### PR TITLE
[IMP] Discuss Sidebar Improvement

### PR DESCRIPTION
**Pull request by qho-odoo**
The commit is to refactor the discuss sidebar

* threads are now organized in categories based on the thread type e.g. chat, channel
* categories can be folded or unfolded by clicking the category title
* for active thread, even if the category is folded, it remains under the category title
* for channel category, a new cog button is added to view all channels
* the active indicator bar is removed. The active item now is highlighted with a different background color
* thread avatar is used for livechat, chat and channel instead of thread icon
* for livchat and chat, threads are now sorted by last activity time (pin or message exchange)
* for channel, threads are sorted by name (case insensitive)

task-id: 2440073

### CODE REVIEW

**Comment by seb-odoo** [comment]
Hello, just a first message to let you know you'll have to rebase to fix conflict (both fix to mentions and removal of channel_ids/follower have been merged recently).

File: addons/im_livechat/models/res_users.py [hunk 1]
```diff
@@ -11,6 +11,8 @@ class Users(models.Model):
     _inherit = 'res.users'
 
     livechat_username = fields.Char("Livechat Username", help="This username will be used as your name in the livechat channels.")
+    # Discuss category livechat open status
+    is_category_livechat_open = fields.Boolean("Is category livechat open", default=True)
```
Comment (tde-banana-odoo) [inline addons/im_livechat/models/res_users.py:15]: We have 2M users in odoo... we don't want to add a new column with True by default. Actually we don't want any new column, but if really necessary, should be False for most users.

File: addons/mail/models/res_users.py [hunk 2]
```diff
@@ -31,6 +31,9 @@ class Users(models.Model):
     moderation_channel_ids = fields.Many2many(
         'mail.channel', 'mail_channel_moderator_rel',
         string='Moderated channels')
+    # Discuss category open states
+    is_category_channel_open = fields.Boolean(string='Is category channel open', default=True)
```
Comment (tde-banana-odoo) [inline addons/mail/models/res_users.py:35]: Don't add 2 new columns on res.users with default=True.

**Review by tde-banana-odoo — COMMENTED** [review]
Just looked at python. Adding 3 columns on res.users for such interface details seem overkill to me. We are trying to remove columns and data in DB and simplify models, I feel like this PR is doing the inverse move.

**Comment by tde-banana-odoo** [comment]
Hello,

Looking at it, it seems you are adding some "interface" configuration for users. As it targets internal users, why not storing it into an external table ? That way you don't bloat the res.users table... which should contain only login-related or security-related information.

**Comment by seb-odoo** [comment]
Discuss interface might (or more than likely will) be extended to all users though. At least in some parts.
I wouldn't mind to store this information in a separate table regardless if we can to keep user purely for login & group.
So we could have a specific table for messaging config. We could move other settings in there too, such as currently existing email/inbox setting, mail bot state? (though that's mostly onboarding stuff that belongs with nothing else), and very likely other new settings to come.
I guess that's something else to add to our "must decide" list.

**Comment by qho-odoo** [comment]
The PR is closed because it is refactored in an another branch.
Check #70986

**Pull request closed without merging**